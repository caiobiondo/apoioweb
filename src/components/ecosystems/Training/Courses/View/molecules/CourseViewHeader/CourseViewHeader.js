import React, { Component } from 'react';
import { Row, Col } from 'react-flexbox-grid';
import { Icon } from 'natura-ui';
import { withRouter } from 'react-router-dom';
import { translate } from 'locale';
import { ROUTE_PREFIX } from 'config';

import {
  Header,
  BackButtonWrapper,
  BackButtonIcon,
  BackButtonText,
} from './CourseViewHeader.styles';

export class CourseViewHeader extends Component {
  goBack = () => {
    this.props.history.push(`${ROUTE_PREFIX}/training/courses`);
  };

  render() {
    return (
      <Header>
        <Row>
          <Col md={12} sm={12}>
            <BackButtonWrapper onClick={this.goBack}>
              <BackButtonIcon>
                <Icon file="ico_back" />
              </BackButtonIcon>

              <BackButtonText>{translate('trainings')}</BackButtonText>
            </BackButtonWrapper>
          </Col>
        </Row>
      </Header>
    );
  }
}

export default withRouter(CourseViewHeader);
