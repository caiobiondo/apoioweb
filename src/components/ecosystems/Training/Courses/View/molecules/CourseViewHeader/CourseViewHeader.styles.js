import styled from 'styled-components';
import { gray890, orange100 } from 'styles/colors';
import { fs16, fs20, RobotoLight } from 'styles/typography';

export const Header = styled.div`
  margin: 15px 0 35px;
`;

export const BackButtonWrapper = styled.div`
  cursor: pointer;
  position: relative;
  text-decoration: none;
`;

export const BackButtonText = styled.span`
  color: ${gray890};
  font-family: ${RobotoLight};
  font-size: ${fs20};
  font-weight: ${fs16};
  padding-left: 50px;
`;

export const BackButtonIcon = styled.span`
  left: 0;
  position: absolute;
  top: 2px;
  width: 20px;

  svg {
    fill: ${orange100};
  }
`;
