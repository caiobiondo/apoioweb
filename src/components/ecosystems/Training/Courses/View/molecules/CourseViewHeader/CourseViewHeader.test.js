import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import CourseViewHeader from './CourseViewHeader';

describe('CourseViewHeader molecules', () => {
  it('should render a menu to training page', () => {
    const renderer = new ShallowRenderer();

    renderer.render(<CourseViewHeader />);
    const result = renderer.getRenderOutput();

    expect(result).toMatchSnapshot();
  });
});
