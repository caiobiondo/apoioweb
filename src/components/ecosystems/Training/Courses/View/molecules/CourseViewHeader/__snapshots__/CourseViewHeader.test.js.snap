// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CourseViewHeader molecules should render a menu to training page 1`] = `
<CourseViewHeaderstyles__Header>
  <Row>
    <Col
      md={4}
      sm={12}
    >
      <CourseViewHeaderstyles__BackButtonWrapper>
        <Link
          replace={false}
          to="/training/categories"
        >
          <CourseViewHeaderstyles__BackButtonIcon>
            <Icon
              file="ico_back"
            />
          </CourseViewHeaderstyles__BackButtonIcon>
          <CourseViewHeaderstyles__BackButtonText>
            Treinamentos
          </CourseViewHeaderstyles__BackButtonText>
        </Link>
      </CourseViewHeaderstyles__BackButtonWrapper>
    </Col>
  </Row>
</CourseViewHeaderstyles__Header>
`;
